xclim/indices/generic.py

```diff
@@ -808,3 +813,107 @@ def degree_days(tas: xr.DataArray, threshold: str, op: str) -> xr.DataArray:
         raise NotImplementedError(f"Condition not supported: '{op}'.")
 
     return to_agg_units(out, tas, op="delta_prod")
+
+
+def first_day_above(
```

Comment:
To go one step further into generification you could have instead a `first_day_threshold_reached` with an operator parameter  taking either `>` or `<` as a valid value.
It would reduce the duplication of code and doc.
This would also ease the adding of new operators such as `>=`, `<=`, `==`
It similar to what is done on `degree_days` for example.

Reply:
Absolutely! Will make changes. Thanks!